torch/_dynamo/guards.py

```diff
@@ -734,6 +739,19 @@ def TENSOR_MATCH(self, guard: Guard, value=None):
                 self.tensor_check_examples.append(value)
                 self.tensor_check_guards.append(guard)
 
+            # Nested Tensor ctx - "ragged_size" is symint placeholder, "requires_grad" is guarded upon
+            if is_traceable_wrapper_subclass(value) and not isinstance(
+                value, NestedTensor
+            ):
+                ctx = value.__tensor_flatten__()[1]  # type: ignore[attr-defined]
+                if ctx is not None:
+                    # Assume that the ctx obeys object equality
+                    obj_store = self.get("G['___stored_objs_by_id']")
```

Comment:
At some point (somewhere) we should add an assert that `ctx` never holds any symbolic types ([list of] Tensor/SymInt/SymFloat/SymBool). Probably not here though, if this is hot-path code.